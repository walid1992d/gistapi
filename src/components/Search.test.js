import { fireEvent, getByTestId, render } from "@testing-library/react";
import { UsernameContext } from "../contexts/usernameContext";
import Search from "./Search";
const setUsernameMock = jest.fn();

describe("Search Component", ()=> {
    
    it("should render correctly", () => {
        
        expect(getContainer()).toMatchSnapshot();
    });
    describe("Input Field", ()=> {
        it("should call setUserName on Enter Key Up", () => {
            const container = getContainer();
            const input = getByTestId(container, "searchInput");
            input.value="12345";
            fireEvent.keyUp(input, {key: 'Enter'});
            expect(setUsernameMock).toHaveBeenCalledWith("12345");
        });
        it("should not call setUserName on Key Up other then enter", () => {
            const container = getContainer();
            const input = getByTestId(container, "searchInput");
            input.value="12345";
            fireEvent.keyUp(input, {key: 'q'});
            expect(setUsernameMock).not.toHaveBeenCalled();
        });
    })
})

const getContainer = ()=> {
    const {container} = render(
        <UsernameContext.Provider value={{usernameState: '', setUserName: setUsernameMock}}>
            <Search></Search>
        </UsernameContext.Provider>
    )
    return container;
}