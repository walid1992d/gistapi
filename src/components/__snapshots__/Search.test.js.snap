// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should render correctly 1`] = `
<div>
  <div
    class="sc-bdnxRM fVdoXT"
  >
    <div
      class="sc-gtsrHT bHTyut"
    >
      <span
        class="octicon octicon-search"
      />
      <input
        class="sc-dlnjwi fozKkc"
        data-testid="searchInput"
        placeholder="Search Gists for the username"
      />
    </div>
  </div>
</div>
`;
